src/nvapi_gpu.cpp

```diff
@@ -372,6 +434,63 @@ extern "C" {
         }
     }
 
+    static NV_THERMAL_TARGET MapThermalTarget(nvmlThermalTarget_t target) {
```

Comment:
Thanks for start using static_assert. Based on your code I did the following for another place https://github.com/jp7677/dxvk-nvapi/blob/master/src/sysinfo/vulkan.cpp#L67
I suggest to move this mapping to a new method `Nvml::ToNvThermalTarget.`